## O primeiro passo para responder as perguntas é combinar todas as tabelas fornecidas.

In [146]:
# Importando as bibliotecas necessárias
import pandas as pd 
import numpy as np

### Primeiramente, é necessário carregar os dados:

In [147]:
base_lojas = pd.read_csv("./data/Base_Lojas.txt", encoding="latin1", sep=";", index_col="Código da Loja", decimal=",")
base_lojas_complementar = pd.read_csv("./data/Base_Lojas complementar.txt", encoding="latin1", sep=";", index_col="Código da Loja", decimal=",")
itens_clientes = pd.read_csv("./data/Itens e clientes por loja.txt", encoding="latin1", sep=";", index_col="Código da Loja", decimal=",")
rh_loja =  pd.read_csv("./data/RHLoja.txt", encoding="latin1", sep=";", index_col="Código da Loja", decimal=",")

### Podemos observar uma parte de cada uma das tabelas:

In [148]:
base_lojas.head()

Formato/ BU Estado
Código da Loja                      
47                 Super Loja     AL
68                 Super Loja     AL
2                  Super Loja     BA
363             Loja do sabor     BA
40                 Super Loja     BA

In [149]:
base_lojas_complementar.head()

Área da Loja (Piso de Vendas, em Metros Quadrados)  \
Código da Loja                                                      
691                                                            61   
656                                                            66   
516                                                            74   
517                                                            75   
623                                                            81   

               Quantidade de SKUs da Loja Possui limpeza Terceirizada?  
Código da Loja                                                          
691                                5.572                           NÃO  
656                                3.192                           NÃO  
516                                3.111                           NÃO  
517                                3.200                           NÃO  
623                                3.194                           NÃO

In [150]:
itens_clientes.head()

Itens Vendidos Clientes atendidos
Código da Loja                                  
722                  148.594             30.593 
723                  208.527             42.021 
701                  230.222             86.470 
756                  244.797             47.141 
760                  267.100             67.828

In [151]:
rh_loja.head()

HC Ativo Total da Loja Horas Trabalhadas - Total da Loja  \
Código da Loja                                                             
509                                  5                            1.064    
500                                  5                            1.078    
194                                 48                           10.293    
160                                 42                            9.153    
398                                 35                            7.672    

               Taxa de Tunover   
Código da Loja                   
509                        2,1%  
500                        2,8%  
194                        2,9%  
160                        3,1%  
398                        3,6%

### O primeiro ponto a ser destacado é que todas as tabelas se relacionam pela chave "Código da Loja". Sendo assim, podemos combiná-las a partir dessa chave.

In [152]:
df = base_lojas.join(base_lojas_complementar, on="Código da Loja", how="outer").join(rh_loja, on="Código da Loja", how="outer").join(itens_clientes, on="Código da Loja", how="outer")

In [153]:
# É convenite ordenar a tabela pelo Código da Loja
df = df.sort_values(by="Código da Loja")

In [154]:
# Podemos ver como fica a tabela final:
df

Formato/ BU Estado  \
Código da Loja                            
1                     Super Loja     MS   
2                     Super Loja     BA   
3                     Super Loja     PI   
4                     Super Loja     MS   
5                     Super Loja     SP   
...                          ...    ...   
756             Lojas da Familia     PE   
757             Lojas da Familia     SP   
759             Lojas da Familia     SP   
760             Lojas da Familia     SP   
761             Lojas da Familia     SP   

               Área da Loja (Piso de Vendas, em Metros Quadrados)  \
Código da Loja                                                      
1                                                           3.586   
2                                                           4.496   
3                                                           4.107   
4                                                           3.523   
5                                                           4.499   
...                                                           ...   
756                                                           140   
757                                                           165   
759                                                           220   
760                                                           132   
761                                                           167   

               Quantidade de SKUs da Loja Possui limpeza Terceirizada?  \
Código da Loja                                                           
1                                 26.904                           SIM   
2                                 26.592                           SIM   
3                                 28.476                           SIM   
4                                 28.669                           SIM   
5                                 32.325                           SIM   
...                                   ...                          ...   
756                                4.881                           NÃO   
757                                4.954                           NÃO   
759                                5.917                           NÃO   
760                                4.935                           NÃO   
761                                5.538                           NÃO   

                HC Ativo Total da Loja Horas Trabalhadas - Total da Loja  \
Código da Loja                                                             
1                                  101                           21.808    
2                                  121                           25.690    
3                                  178                           39.144    
4                                  111                           23.790    
5                                  247                           52.062    
...                                ...                               ...   
756                                  9                            1.890    
757                                  9                            1.891    
759                                 10                            2.287    
760                                  7                            1.560    
761                                  8                            1.788    

               Taxa de Tunover  Itens Vendidos Clientes atendidos  
Código da Loja                                                     
1                         45,5%     5.662.916            585.950   
2                         27,3%    10.006.477            837.415   
3                         29,3%    13.158.512          1.092.452   
4                         40,0%     6.281.119            755.513   
5                         17,3%    20.743.582          2.008.825   
...                         ...            ...                ...  
756                        4,4%       244.797             47.141   

In [156]:
df.to_csv("./data/dados_rd.txt")

### 1) a) Qual a loja mais produtiva da base disponibilizada?

In [131]:
# É preciso converter o tipo:
df["Itens Vendidos"] = df["Itens Vendidos"].str.replace('.', '')
df["Itens Vendidos"] = df["Itens Vendidos"].astype('float')
df["Horas Trabalhadas - Total da Loja"] = df["Horas Trabalhadas - Total da Loja"].str.replace('.', '')
df["Horas Trabalhadas - Total da Loja"] = df["Horas Trabalhadas - Total da Loja"].astype('float')

In [133]:
# Criando a coluna produtividade:
df["produtividade"] = df["Itens Vendidos"]/df["Horas Trabalhadas - Total da Loja"]

In [139]:
# Buscando pela loja mais produtiva
df.loc[df.produtividade == df.produtividade.max()]

Formato/ BU Estado  \
Código da Loja                           
151             Lojas Esperança     SP   

               Área da Loja (Piso de Vendas, em Metros Quadrados)  \
Código da Loja                                                      
151                                                           943   

               Quantidade de SKUs da Loja Possui limpeza Terceirizada?  \
Código da Loja                                                           
151                                9.689                           NÃO   

                HC Ativo Total da Loja  Horas Trabalhadas - Total da Loja  \
Código da Loja                                                              
151                                 35                             7585.0   

               Taxa de Tunover   Itens Vendidos Clientes atendidos  \
Código da Loja                                                       
151                       12,3%       6424913.0           485.832    

                produtividade  
Código da Loja                 
151                847.055109


### Número da loja mais produtiva: 151

1) b) Qual é a média de proditividade das lojas que possuem limpeza terceirizada?
df.loc[df[]]

In [142]:
df.loc[df["Possui limpeza Terceirizada?"] == "SIM", "produtividade"].mean()

413.9929708324899

1) c) Qual a média de horas trabalhadas das lojas "Super Loja"?

In [144]:
df.loc[df["Formato/ BU"] == "Super Loja", "Horas Trabalhadas - Total da Loja"].mean()

26120.378151260506

Podemos exportar os dados calculados para resolver as demais questões no tableau:

In [145]:
df.to_csv("./data/dados_teste_rd")